### ISSUE TITLE
OverridingException: Binding bind<LayoutInflater>() with ? { ? } must not override an existing binding

### ISSUE THREAD

**Issue by Gorcyn** [SalomonBrys/Kodein#58]
Hi!

I tried to implement Kodein in a project and met a crash.
I started investigating... then I gave up.

Then I could reproduce it easily with the Kodein master branch.

I edited the **android-demo** sample project dependencies replacing:
```
  compile project(':kodein-core')
  compile project(':kodein')
  compile project(':kodein-android')
```
with:
```
  final KODEIN_VERSION = '3.2.0'
  compile "com.github.salomonbrys.kodein:kodein-core:$KODEIN_VERSION"
  compile "com.github.salomonbrys.kodein:kodein:$KODEIN_VERSION"
  compile "com.github.salomonbrys.kodein:kodein-android:$KODEIN_VERSION"
```

Am I missing something?

```--------- beginning of crash
03-23 14:09:35.556 22623-22623/kodein.demo E/AndroidRuntime: FATAL EXCEPTION: main
                                                             Process: kodein.demo, PID: 22623
                                                             java.lang.RuntimeException: Unable to start activity ComponentInfo{kodein.demo/kodein.demo.MainActivity}: com.github.salomonbrys.kodein.Kodein$OverridingException: Binding bind<LayoutInflater>() with ? { ? } must not override an existing binding.
                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
                                                                 at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
                                                                 at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                 at android.os.Looper.loop(Looper.java:154)
                                                                 at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
                                                              Caused by: com.github.salomonbrys.kodein.Kodein$OverridingException: Binding bind<LayoutInflater>() with ? { ? } must not override an existing binding.
                                                                 at com.github.salomonbrys.kodein.KodeinContainer$Builder._checkOverrides(KodeinContainer.kt:246)
                                                                 at com.github.salomonbrys.kodein.KodeinContainer$Builder.access$_checkOverrides(KodeinContainer.kt:137)
                                                                 at com.github.salomonbrys.kodein.KodeinContainer$Builder$BindBinder.with(KodeinContainer.kt:297)
                                                                 at com.github.salomonbrys.kodein.Kodein$Builder$TBuilder$TypeBinder.with(Kodein.kt:189)
                                                                 at com.github.salomonbrys.kodein.android.ActivityInjector$initializeInjector$activityModule$1.invoke(KodeinAndroidComponents.kt:85)
                                                                 at com.github.salomonbrys.kodein.android.ActivityInjector$initializeInjector$activityModule$1.invoke(KodeinAndroidComponents.kt:75)
                                                                 at com.github.salomonbrys.kodein.Kodein$Builder.<init>(Kodein.kt:303)
                                                                 at com.github.salomonbrys.kodein.Kodein$Builder.import(Kodein.kt:387)
                                                                 at com.github.salomonbrys.kodein.android.KodeinAndroidComponentsKt$inject$kodein$1.invoke(KodeinAndroidComponents.kt:21)
                                                                 at com.github.salomonbrys.kodein.android.KodeinAndroidComponentsKt$inject$kodein$1.invoke(KodeinAndroidComponents.kt)
                                                                 at com.github.salomonbrys.kodein.Kodein$Builder.<init>(Kodein.kt:303)
                                                                 at com.github.salomonbrys.kodein.internal.KodeinImpl.<init>(KodeinImpl.kt:34)
                                                                 at com.github.salomonbrys.kodein.Kodein$Companion.invoke(Kodein.kt:469)
                                                                 at com.github.salomonbrys.kodein.Kodein$Companion.invoke$default(Kodein.kt:469)
                                                                 at com.github.salomonbrys.kodein.android.KodeinAndroidComponentsKt.inject(KodeinAndroidComponents.kt:18)
                                                                 at com.github.salomonbrys.kodein.android.KodeinAndroidComponentsKt.access$inject(KodeinAndroidComponents.kt:1)
                                                                 at com.github.salomonbrys.kodein.android.ActivityInjector$DefaultImpls.initializeInjector(KodeinAndroidComponents.kt:90)
                                                                 at com.github.salomonbrys.kodein.android.KodeinActivity.initializeInjector(KodeinAndroidComponents.kt:103)
                                                                 at com.github.salomonbrys.kodein.android.KodeinActivity.onCreate(KodeinAndroidComponents.kt:109)
                                                                 at kodein.demo.MainActivity.onCreate(MainActivity.kt:26)
                                                                 at android.app.Activity.performCreate(Activity.java:6679)
                                                                 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726) 
                                                                 at android.app.ActivityThread.-wrap12(ActivityThread.java) 
                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477) 
                                                                 at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                 at android.os.Looper.loop(Looper.java:154) 
                                                                 at android.app.ActivityThread.main(ActivityThread.java:6119) 
                                                                 at java.lang.reflect.Method.invoke(Native Method) 
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) ```

**Comment by SalomonBrys** [comment]
Hi @Gorcyn,
this issue was already reported in Slack (forgot to make an issue about it here).

It is already corrected and will be deployed probably tomorrow with the 3.3.0 release :)

Salomon.

**Comment by ZineeEddine** [comment]
Hello, I am using kodein 4.1.0 in an android mobile app project, with android MVP architecture and using Java and Kotlin.

The version of kotlin is 1.6.

I have the same problem, when I launch the application a crash occurs and I have the same error message.

Can you help me solve the problem?

I note that the problem had to be corrected since version 3.3.0, knowing that I use version 4.1.0


I am waiting for your response, thank you very much
